packages/mds-utils/state-machine.ts

```diff
@@ -1,58 +1,136 @@
+    [VEHICLE_EVENTS.maintenance_pick_up]: [VEHICLE_STATES.removed],
+    [VEHICLE_EVENTS.missing]: [VEHICLE_STATES.unknown],
+    [VEHICLE_EVENTS.on_hours]: [VEHICLE_STATES.available],
+    [VEHICLE_EVENTS.rebalance_pick_up]: [VEHICLE_STATES.removed],
+    [VEHICLE_EVENTS.system_resume]: [VEHICLE_STATES.available],
+    [VEHICLE_EVENTS.unspecified]: [VEHICLE_STATES.available, VEHICLE_STATES.removed]
+  },
+  [VEHICLE_STATES.on_trip]: {
+    [VEHICLE_EVENTS.comms_lost]: [VEHICLE_STATES.unknown],
+    [VEHICLE_EVENTS.trip_cancel]: [VEHICLE_STATES.available],
+    [VEHICLE_EVENTS.trip_end]: [VEHICLE_STATES.available],
+    [VEHICLE_EVENTS.trip_leave_jurisdiction]: [VEHICLE_STATES.elsewhere],
+    [VEHICLE_EVENTS.missing]: [VEHICLE_STATES.unknown]
   },
   [VEHICLE_STATES.removed]: {
-    [VEHICLE_EVENTS.trip_enter]: VEHICLE_STATES.trip,
-    [VEHICLE_EVENTS.provider_drop_off]: VEHICLE_STATES.available,
-    [VEHICLE_EVENTS.deregister]: VEHICLE_STATES.inactive
+    [VEHICLE_EVENTS.agency_drop_off]: [VEHICLE_STATES.available],
+    [VEHICLE_EVENTS.decommissioned]: [VEHICLE_STATES.removed],
+    [VEHICLE_EVENTS.provider_drop_off]: [VEHICLE_STATES.available],
+    [VEHICLE_EVENTS.unspecified]: [VEHICLE_STATES.available]
   },
   [VEHICLE_STATES.reserved]: {
-    [VEHICLE_EVENTS.trip_start]: VEHICLE_STATES.trip,
-    [VEHICLE_EVENTS.cancel_reservation]: VEHICLE_STATES.available
-  },
-  [VEHICLE_STATES.trip]: {
-    [VEHICLE_EVENTS.trip_leave]: VEHICLE_STATES.elsewhere,
-    [VEHICLE_EVENTS.trip_end]: VEHICLE_STATES.available
+    [VEHICLE_EVENTS.comms_lost]: [VEHICLE_STATES.unknown],
+    [VEHICLE_EVENTS.missing]: [VEHICLE_STATES.unknown],
+    [VEHICLE_EVENTS.reservation_cancel]: [VEHICLE_STATES.available],
+    [VEHICLE_EVENTS.trip_start]: [VEHICLE_STATES.on_trip],
+    [VEHICLE_EVENTS.unspecified]: [VEHICLE_STATES.available]
   },
-  [VEHICLE_STATES.unavailable]: {
-    [VEHICLE_EVENTS.service_start]: VEHICLE_STATES.available,
-    [VEHICLE_EVENTS.deregister]: VEHICLE_STATES.inactive,
-    [VEHICLE_EVENTS.agency_pick_up]: VEHICLE_STATES.removed,
-    [VEHICLE_EVENTS.provider_pick_up]: VEHICLE_STATES.removed
+  [VEHICLE_STATES.unknown]: {
+    [VEHICLE_EVENTS.agency_drop_off]: [VEHICLE_STATES.available],
+    [VEHICLE_EVENTS.agency_pick_up]: [VEHICLE_STATES.removed],
+    [VEHICLE_EVENTS.comms_restored]: [
+      VEHICLE_STATES.available,
+      VEHICLE_STATES.elsewhere,
+      VEHICLE_STATES.reserved,
+      VEHICLE_STATES.on_trip,
+      VEHICLE_STATES.non_operational
+    ],
+    [VEHICLE_EVENTS.decommissioned]: [VEHICLE_STATES.removed],
+    [VEHICLE_EVENTS.provider_drop_off]: [VEHICLE_STATES.available],
+    [VEHICLE_EVENTS.unspecified]: [VEHICLE_STATES.available, VEHICLE_STATES.removed]
   }
 }
 
-const getNextState = (currStatus: VEHICLE_STATE, nextEvent: VEHICLE_EVENT): VEHICLE_STATE | undefined => {
+const getNextStates = (currStatus: VEHICLE_STATE, nextEvent: VEHICLE_EVENT): Array<VEHICLE_STATE> | undefined => {
```

Comment:
What does it mean to transition to an array of states?

Reply:
If you have a `comms_lost` event, the vehicle goes into the `unknown` state. Then when `comms_restored` happens, the vehicle could be in pretty much state--`available`, `non_operational`, etc.